runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -416,38 +415,46 @@ function M._enable(bufnr)
     bufnr = api.nvim_get_current_buf()
   end
 
-  if _autocmd_ids[bufnr] then
-    return
+  if not bufstates[bufnr] then
+    bufstates[bufnr] = { enabled = true }
+
+    api.nvim_create_autocmd('LspNotify', {
+      buffer = bufnr,
+      callback = function(opts)
+        if
+          opts.data.method ~= 'textDocument/didChange'
+          and opts.data.method ~= 'textDocument/didOpen'
```

Comment:
It would be nice to be able to use the LSP method names as patterns for the autocommand...

Reply:
Yeah, unfortunately patterns can't be used with buffer local autocmds:
https://neovim.io/doc/user/api.html#api-autocmd